### PR TITLE
Add modifiers to dynamic_bitset, and add constexpr

### PR DESCRIPTION
**Pull request by Alex-PLACET**

### CODE REVIEW

File: include/sparrow/buffer/dynamic_bitset.hpp [hunk 1]
```diff
@@ -388,45 +424,95 @@ namespace sparrow
...
+                "dynamic_bitset_base::at: index out of range for dynamic_bitset_base of size"
+                + std::to_string(size()) + " at index " + std::to_string(pos)
+            );
+        }
+        return test(pos);
+    }
+
+    template <random_access_range B>
+    constexpr auto dynamic_bitset_base<B>::front() -> reference
+    {
```
Comment (Klaim) [inline include/sparrow/buffer/dynamic_bitset.hpp:492]: Add assertions in non-throwing functions that access the elements (front, back, etc.)?

File: include/sparrow/buffer/dynamic_bitset.hpp [hunk 2]
```diff
@@ -561,24 +647,156 @@ namespace sparrow
         zero_unused_bits();
     }
 
+    template <random_access_range B>
+    constexpr void dynamic_bitset_base<B>::clear() noexcept
+    {
+        resize(0);
```
Comment (Klaim) [inline include/sparrow/buffer/dynamic_bitset.hpp:653]: This calls buffer's resize which will call the destructor of the storage type. Just making sure I understand correctly: is this always equivalent to just setting every bytes to zero? (maybe we should constrain the storage type to only be a basic type?)

File: include/sparrow/buffer/dynamic_bitset.hpp [hunk 3]
```diff
@@ -561,24 +647,156 @@ namespace sparrow
...
+        SPARROW_ASSERT_TRUE(cbegin() <= pos);
+        SPARROW_ASSERT_TRUE(pos <= cend());
+        const auto index = static_cast<size_type>(std::distance(cbegin(), pos));
+        const size_type old_size = size();
+        const size_type count = static_cast<size_type>(std::distance(first, last));
+        const size_type new_size = old_size + count;
+
+        resize(new_size);
+
+        // TODO: The current implementation is not efficient. It can be improved.
```
Comment (Klaim) [inline include/sparrow/buffer/dynamic_bitset.hpp:705]: Consider creating an issue about this type's performance once this is merged?

**Review by Klaim — APPROVED** [review]
LGTM, just needs more assertions 👍🏼

File: include/sparrow/buffer/dynamic_bitset.hpp (hunk 3 above)
Comment (Alex-PLACET) [reply include/sparrow/buffer/dynamic_bitset.hpp:705]: Ok added: #201

File: include/sparrow/buffer/dynamic_bitset.hpp (hunk 2 above)
Comment (Alex-PLACET) [reply include/sparrow/buffer/dynamic_bitset.hpp:653]: I modified the code and is now:
```
m_buffer.clear();
m_size = 0;
m_null_count = 0;
```

File: include/sparrow/buffer/dynamic_bitset.hpp (hunk 1 above)
Comment (Alex-PLACET) [reply include/sparrow/buffer/dynamic_bitset.hpp:492]: done

**Review by JohanMabille — APPROVED** [review]

**Pull request merged**